### ISSUE TITLE
fix: mason_logger is not working properly when it reaches to the end of terminal

### ISSUE THREAD

**Issue by payam-zahedi** [felangel/mason#1356]
**Description**

When we are using choose options and cursor is in the end of the terminal screen its not working properly.

Tryed on: Windows, Linux, macOS

**Steps To Reproduce**

1. try to run the following code 
2. You will see in the video above it's not properly

**Screenshots**

https://github.com/felangel/mason/assets/47558577/04e04a46-4a09-4be4-b28d-10accf46415a

**Comment by payam-zahedi** [comment]
@felangel Do you have any Idea what is happening and why?

I can try to fix it if you can help me

**Comment by payam-zahedi** [comment]
Probably related to #507

**Comment by felangel** [comment]
> I had an investigation on this
> 
> So I think the problem is related to the save cursor line and the behavior of console
> 
> it seems when the cursor is at the last line, and we save the cursor position
> 
> and after that we have some new line (scrolling) the cursor position will stay the same
> 
> Basically, cursor is not respecting the scrolling.

Yup haven’t had time to look at how other tools overcome this problem but happy to review any PRs if you get to it first 👍

**Comment by payam-zahedi** [comment]
I have some Idea. 

Will make a PR

**Comment by MarkOSullivan94** [comment]
Also noticed this issue

**Comment by felangel** [comment]
@MarkOSullivan94 @payam-zahedi are you still able to reproduce with the latest version of `mason_logger`?

**Comment by payam-zahedi** [comment]
@felangel 

I can not reproduce it anymore on the example.
It seems it is fixed now. 

but I will try to test it on our product to make sure about it.


Can you please give me the link of commit where you fixed it I could not find it.

**Comment by mugbug** [comment]
Also experiencing this. One workaround I've found is to clear the terminal before calling `chooseOne` or `chooseAny`. Something like this:

```dart
const clearTerminal = '\x1B[2J\x1B[0;0H';
_logger.info(clearTerminal);
```

**Comment by felangel** [comment]
> Also experiencing this. One workaround I've found is to clear the terminal before calling `chooseOne` or `chooseAny`. Something like this:
> 
> ```dart
> const clearTerminal = '\x1B[2J\x1B[0;0H';
> _logger.info(clearTerminal);
> ```

That would work but I'm not sure it's a good idea to delete parts of the terminal buffer without the users' permission. It might be annoying/undesirable to have important output deleted just because of the choice prompt 🤔

**Comment by felangel** [comment]
This has been fixed in mason_logger v0.3.6
